beluga-mqtt/src/manager.rs

```diff
@@ -0,0 +1,86 @@
+use std::collections::{HashMap, HashSet};
+
+use rumqttc::Publish;
+use tokio::sync::broadcast::{self, Receiver, Sender};
+
+#[derive(Debug, Default)]
+pub(super) struct SubscriberManager {
```

Comment:
This type definitely deserves unit tests especially surrounding the edge cases.

I see for instance that you already thought of the case where someone re-subscribes to a topic before the previous "unsubscribed" happens, but it would be great to have this interaction as a unit test as well.